syntax = "proto3";
package l2db;
import "common.proto";
import "mysql_database_table.proto";
option cc_generic_services = true;

message LoginRequest
{
  string account = 1;
  string password = 2;
}

message LoginResponse
{
	ClientError error = 1;
  account_database account_player = 2;
}

message CreatePlayerRequest
{
}

message CreatePlayerRespone
{
  ClientError error = 1;
  account_database account_player = 2;
  uint64 player_id = 3;
}

message EnterGameRequest
{
  
}

message EnterGameRespone
{
  ClientError error = 1;
  uint64 player_id = 2;
}

service LoginService {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc CratePlayer (CreatePlayerRequest) returns (CreatePlayerRespone);
  rpc EnterGame (EnterGameRequest) returns (EnterGameRequest);
}
